main.go

```diff
@@ -54,31 +55,35 @@ func main() {
 		os.Exit(1)
 	}
 
-	s, err := b.Build(dc)
+	dsn, err := b.Build(dc)
 
 	if err != nil {
 		fmt.Println("Error occurred building connection string:", err)
 		os.Exit(1)
 	}
 
-	dc.URL = s
 	client := NewClient(config.GeckoboardAPIKey)
+	db, err := newDBConnection(dc.Driver, dsn)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
 
```

Comment:
Its a thing - I read so many places that its rare we should need to close - but I do worry that we should handle it in the following cases.

1. Where the processAllDatasets isn't setup to push every x seconds we should definitely close
2. When we have the connection pool and someone SIGTERM the program - we probably should also be monitoring to cleanly close any connections

but for point 2 I read that we shouldn't need to weirdly and only for mongo it is very important to close.